test/e2e/test_system.py

```diff
@@ -4,603 +4,579 @@
 
+        self.patch_system(system_id, {"parent_id": parent_id})
+        self.check_patch_system_failed_with_message(
+            409, "A System with the same name already exists within the parent System"
+        )
 
-def test_get_system_breadcrumbs_when_trail_length_less_than_maximum(test_client):
-    """
-    Test getting the breadcrumbs for a system with less than the the maximum trail length
-    """
-    systems = _post_n_systems(test_client, BREADCRUMBS_TRAIL_MAX_LENGTH - 1)
+    def test_partial_update_parent_id_to_child_of_self(self):
+        """Test updating the parent_id of a System to one of its own children"""
 
-    # Get breadcrumbs for last added
-    response = test_client.get(f"/v1/systems/{systems[-1]['id']}/breadcrumbs")
+        system_ids = self.post_nested_systems(2)
+        self.patch_system(system_ids[0], {"parent_id": system_ids[1]})
+        self.check_patch_system_failed_with_message(422, "Cannot move a system to one of its own children")
 
-    assert response.status_code == 200
-    assert response.json() == {"trail": [[system["id"], system["name"]] for system in systems], "full_trail": True}
+    def test_partial_update_parent_id_to_non_existent(self):
+        """Test updating the parent_id of a System to a non-existent System"""
 
+        system_id = self.post_system(SYSTEM_POST_DATA_ALL_VALUES_NO_PARENT)
+        self.patch_system(system_id, {"parent_id": str(ObjectId())})
+        self.check_patch_system_failed_with_message(422, "The specified parent System does not exist")
 
-def test_get_system_breadcrumbs_when_trail_length_maximum(test_client):
-    """
-    Test getting the breadcrumbs for a system with the maximum trail length
-    """
-    systems = _post_n_systems(test_client, BREADCRUMBS_TRAIL_MAX_LENGTH)
+    def test_partial_update_parent_id_to_invalid(self):
+        """Test updating the parent_id of a System to an invalid id"""
 
-    # Get breadcrumbs for last added
-    response = test_client.get(f"/v1/systems/{systems[-1]['id']}/breadcrumbs")
+        system_id = self.post_system(SYSTEM_POST_DATA_ALL_VALUES_NO_PARENT)
+        self.patch_system(system_id, {"parent_id": "invalid-id"})
+        self.check_patch_system_failed_with_message(422, "The specified parent System does not exist")
 
-    assert response.status_code == 200
-    assert response.json() == {"trail": [[system["id"], system["name"]] for system in systems], "full_trail": True}
+    def test_partial_update_name_to_duplicate(self):
+        """Test updating the name of a System to conflict with a pre-existing one"""
 
+        self.post_system(SYSTEM_POST_DATA_REQUIRED_VALUES_ONLY)
+        system_id = self.post_system(SYSTEM_POST_DATA_ALL_VALUES_NO_PARENT)
+        self.patch_system(system_id, {"name": SYSTEM_POST_DATA_REQUIRED_VALUES_ONLY["name"]})
+        self.check_patch_system_failed_with_message(
+            409, "A System with the same name already exists within the parent System"
+        )
```

Comment:
Also worth mentioning things like this I don't think are in the spirit of DSL, as really it would require additional functions with names such as

self.create_system_with_child()
self.move_system_to_previously_created_system()

But I thought that going that far is where the disadvantages of the method start to occur as maintenance becomes more awkward so I thought this could be a decent hybrid instead.